yt_dlp/extractor/netverse.py

```diff
@@ -0,0 +1,174 @@
+from .common import InfoExtractor
+from ..utils import (
+    traverse_obj,
+    ExtractorError,
+    # GeoRestrictedError,
+)
+
+
+class NetverseBaseIE(InfoExtractor):
+    def get_required_json(self, url, data=None):
+        match = self._match_valid_url(url).groupdict()
+        display_id, sites_type = match['display_id'], match['type']
+
+        if sites_type == 'watch' or sites_type == 'video':
+            media_api_url = f'https://api.netverse.id/medias/api/v2/watchvideo/{display_id}'
+        elif sites_type == 'webseries':
+            media_api_url = f'https://api.netverse.id/medias/api/v2/webseries/{display_id}'
+
+        json_data = self._download_json(media_api_url, display_id, data=data)
+
+        if json_data.get('error'):
+            raise ExtractorError(json_data.get('message'))
+
+        return json_data
+
+    def get_access_id(self, dailymotion_url):
+        return dailymotion_url.split('/')[-1]
+
+    def _call_metadata_api_from_video_url(self, dailymotion_url):
+        access_id = self.get_access_id(dailymotion_url)
+        metadata_json = self._call_metadata_api(access_id)
+        return access_id, metadata_json
+
+    def _call_metadata_api(self, access_id, req_file_type='video', query=None):
+        video_metadata_api_url = f'https://www.dailymotion.com/player/metadata/{req_file_type}/{access_id}'
+
+        if query is None:
+            required_query = {
+                'embedder': 'https://www.netverse.id'
+            }
+        else:
+            required_query = query
+
+        return self._download_json(video_metadata_api_url, access_id, query=required_query)
```

Comment:
```suggestion
        return self._download_json(
            f'https://www.dailymotion.com/player/metadata/{req_file_type}/{access_id}',
            access_id, query=query or {'embedder': 'https://www.netverse.id'})
```

This again brings into question whether this should be a separate function in the first place

Reply:
The function got refactored, done